### PR TITLE
Review and correct //test/exe:* targets for Windows

### PR DESCRIPTION
**Pull request by sunjayBhatia**
Commit Message: Review and correct //test/exe:* targets for Windows
Additional Description:
- Capture the Clean vs Modified scm build status on Windows
- Corrects the version_out test
- on MacOS, override the Clean vs Modified with an explicit argument
  to preserve existing behavior
- Other changes are editorial to justify the reason for skipping tests

Risk Level: Low
Testing: Enables a couple exe tests on Windows, N/A otherwise
Docs Changes: N/A
Release Notes: N/A

### CODE REVIEW

**Comment by sunjayBhatia** [comment]
cc @wrowe

**Comment by wrowe** [comment]
There is an open question that need not be addressed by this patch, but is worthy of discussion.

Was substituting 'Library' for the boolean 'Clean|Modified' state of the given envoy source branch a desirable toggle, in the general case? It was occurring for iOS by design, but also Windows and any other use of the --manual_stamp toggle.

Could '/Library' be another "facility" tacked onto the version along with SSL library elections, preserving the Clean|Modified state even for iOS?

Is the manual_stamp toggle a reasonable cause to inject that 'Library' token? Or should that be an alternate/additional toggle?

This proposed PR is identical to existing manual_stamp behavior on both OS/X (which doesn't use it, so the patch is a no-op) and iOS (which does use it, and historically chooses to substitute 'Library'.) It is improved behavior for Windows which shares the manual_stamp creation.

**Comment by sunjayBhatia** [comment]
@lizan this is ready for review when you get a chance

File: source/common/common/BUILD [hunk 1]
```diff
@@ -348,7 +348,11 @@ genrule(
 genrule(
     name = "generate_version_linkstamp",
     outs = ["lib/version_linkstamp.h"],
-    cmd = "$(location :generate_version_linkstamp.sh) >> $@",
+    cmd = select({
+        # Only iOS builds typically follow this logic, OS/X is built as a normal binary
+        "//bazel:apple": "$(location :generate_version_linkstamp.sh) Library >> $@",
```
Comment (lizan) [inline source/common/common/BUILD:353]: @junr03 to double check

Comment (wrowe) [reply source/common/common/BUILD:353]: Good morning. Please note that this PR blocks adoption of RBE of //test/... on windows, and is becoming more of a showstopper for us at 2 weeks stale.

So for background, we cautiously adopted a neutral behavior for this patch for iOS mobile, this PR is a no-op to the existing behavior on that platform. (Library) remains (Library) at this time on apple (non-10.x) targets.

https://github.com/envoyproxy/envoy/pull/11291#issuecomment-632827626 is simply a discussion topic to consider if we would want to make this consistent on iOS mobile, but this does not need to be changed at this time via this PR.

**Comment by mattklein123** [comment]
Please merge master once the OSX CI issue is fixed.

/wait

**Pull request merged**